data/scripts/variants_spdi.py

```diff
@@ -0,0 +1,124 @@
+import gzip
+import csv
+import hashlib
+import os
+from ga4gh.vrs.extras.translator import Translator
+from ga4gh.vrs.dataproxy import create_dataproxy
+
+chr_map = {
+    '1': 'NC_000001.11',
+    '2': 'NC_000002.12',
+    '3': 'NC_000003.12',
+    '4': 'NC_000004.12',
+    '5': 'NC_000005.10',
+    '6': 'NC_000006.12',
+    '7': 'NC_000007.14',
+    '8': 'NC_000008.11',
+    '9': 'NC_000009.12',
+    '10': 'NC_000010.11',
+    '11': 'NC_000011.10',
+    '12': 'NC_000012.12',
+    '13': 'NC_000013.11',
+    '14': 'NC_000014.9',
+    '15': 'NC_000015.10',
+    '16': 'NC_000016.10',
+    '17': 'NC_000017.11',
+    '18': 'NC_000018.10',
+    '19': 'NC_000019.10',
+    '20': 'NC_000020.11',
+    '21': 'NC_000021.9',
+    '22': 'NC_000022.11',
+    'X': 'NC_000023.11',
+    'Y': 'NC_000024.10'
+
+}
+
+
+def build_variant_id(chr, pos_first_ref_base, ref_seq, alt_seq, assembly='GRCh38'):
+    key = '{}_{}_{}_{}_{}'.format(str(chr).replace(
+        'chr', '').lower(), pos_first_ref_base, ref_seq, alt_seq, assembly)
+    return hashlib.sha256(key.encode()).hexdigest()
+
+# the spdi generated by vrs is in this format: SEQUENCE-ID:POSITION:DEL-LEN:INS-SEQUENCE
+# need to convert to this format for igvf catalog: SEQUENCE-ID:POSITION:DEL-SEQUENCE:INS-SEQUENCE
+
+
+def convert_spdi(spdi, seq):
+    ls = spdi.split(':')
+    ls[2] = seq
+    spdi = ':'.join(ls)
+    return spdi
+
+
+# in order to use translator locally, need to install seqrepo and pull data to local first
+# check instruction here: https://github.com/biocommons/biocommons.seqrepo
+# for each file, an output file will be genereated.
+# each row in output file cotains: id in catalog, chr, pos, ref, alt, spdi, hgvs
+
+
+def main():
+    dp = create_dataproxy('seqrepo+file:///usr/local/share/seqrepo/2018-11-26')
+    translator = Translator(data_proxy=dp)
+    input_folder = '/home/ubuntu/datasets/favor'
+    ouput_folder = '/home/ubuntu/datasets/favor_output'
```

Comment:
this only works on a node instance, should we make this configurable so we can run it locally?

Reply:
another refactoring tip is that all hard coded strings should be constants

Reply:
You can run it locally. I have comment in the script regarding the instruction to set it up locally. I will use args to pass in the input and output file